include/PrimeTower.h

```diff
@@ -100,11 +103,47 @@ class PrimeTower
      */
     void subtractFromSupport(SliceDataStorage& storage);
 
+    /*!
+     * Get the outer polygon for the given layer, which may be the priming polygon only, or a larger polygon for layers with a base
+     *
+     * \param[in] layer_nr The index of the layer
+     * \return The outer polygon for the prime tower at the given layer
+     */
+    const Polygons& getOuterPoly(const LayerIndex& layer_nr) const;
+
+    /*!
+     * Get the outer polygon for the very first layer, which may be the priming polygon only, or a larger polygon if there is a base
+     */
+    const Polygons& getGroundPoly() const;
+
 private:
+    /*!
+     * \see PrimeTower::generatePaths
+     *
+     * Generate extra rings around the actual prime rings for a stronger base
+     *
+     * \param inset The inner circle of the rings to start generating the rings from
+     * \param rings The number of rings to add
+     * \param line_width The actual line width to distance the rings from each other
+     * \return The generated rings paths
+     */
+    static ExtrusionMoves generatePaths_base(const Polygons& inset, size_t rings, coord_t line_width);
```

Comment:
This generates a number of inset paths? Seems generically useful geometric function, does it belong in a geometry toolbox?

Reply:
That's the very same thing that the brim algorithm does indeed. I'll see if I can merge them.

Reply:
Didn't merge with the brim, but at least they are now in PolygonUtils